src/value/utils.rs

```diff
@@ -0,0 +1,254 @@
+#[cfg(test)]
```

Comment:
I'm not sure I follow this file — what is it testing? It seems to also contain a bunch of implementation, but that's only used for tests? If this is an orthogonal change, would you mind making it in a separate PR?

Reply:
Here's some code to help the tests run correctly, migrated from the old mysql_common. I don't think this is considered orthogonal :(

Reply:
If this comes from `mysql_common`, why move it in here? Does it test anything that actually relates to this crate? Why was it removed from `mysql_common` in the first place?

Reply:
This is because `mysql_common` removed functions like `write_bin_value`. In fact, this is just to ensure that the existing tests work properly.

It looks like `mysql_common` is now updated with serialization/deserialization.

Reply:
Hmm, I don't think they actually removed those functions, they just renamed them. For example, decimal decoding is now in [`Decimal::parse_bytes`](https://docs.rs/mysql_common/0.27.4/mysql_common/binlog/decimal/struct.Decimal.html#method.parse_bytes). I think we should just switch to those methods instead of inlining the old implementation (which may have been replaced because it was buggy).